stock_picking_manual_procurement_group/tests/test_stock_picking.py

```diff
@@ -0,0 +1,198 @@
+        route = self.env[
+            "stock.location.route"].create({
+                "name": "Inter-Warehouse Pull",
+                "product_selectable": True,
+                })
+        self.rule = self.env[
+            "procurement.rule"].create({
+                "name": "Pull from Main WH",
+                "route_id": route.id,
+                "location_id": self.loc_transit_1.id,
+                "location_src_id": self.env.ref(
+                    "stock.warehouse0").lot_stock_id.id,
+                "procure_method": "make_to_order",
+                "action": "move",
+                "picking_type_id": self.env.ref(
+                    "stock.warehouse0").int_type_id.id,
+                })
+
+    def _create_picking(self, create_procurement_group=False):
+        res_picking = {
+            "picking_type_id": self.env.ref(
+                "stock.warehouse0").int_type_id.id,
+            "create_procurement_group": create_procurement_group,
+            }
+
+        picking = self.env[
+            "stock.picking"].create(res_picking)
+
+        res_move_1 = {
+            "name": self.product1.name,
+            "product_id": self.product1.id,
+            "picking_id": picking.id,
+            "product_uom_qty": 10.0,
+            "product_uom": self.product1.uom_id.id,
+            "procure_method": "make_to_order",
+            "location_id": self.loc_transit_1.id,
+            "location_dest_id": self.env.ref(
+                "stock.stock_warehouse_shop0").lot_stock_id.id,
+            }
+
+        self.env["stock.move"].create(res_move_1)
+
+        res_move_2 = {
+            "name": self.product2.name,
+            "product_id": self.product2.id,
+            "picking_id": picking.id,
+            "product_uom_qty": 10.0,
+            "product_uom": self.product1.uom_id.id,
+            "procure_method": "make_to_order",
+            "location_id": self.loc_transit_1.id,
+            "location_dest_id": self.env.ref(
+                "stock.stock_warehouse_shop0").lot_stock_id.id,
+            }
+
+        self.env["stock.move"].create(res_move_2)
+
+        return picking
+
+    def _check_move_group(self, picking):
+        for move in picking.move_lines:
+            self.assertEqual(
+                move.group_id,
+                picking.group_id)
+
+    def _get_generated_procurements(self, picking):
+        criteria = [
+            ("group_id", "=", picking.group_id.id),
+            ]
+        return self.obj_procurement.search(criteria)
+
+    def _run_procurements(self, procurements):
+        for procurement in procurements:
+            procurement.write(
+                {"rule_id": self.rule.id})
+
+            if procurement.state == "confirmed":
+                procurement.run()
+
+    def _get_new_moves(self, procurements):
+        new_moves = []
+        for procurement in procurements:
+            new_moves += procurement.move_ids
+
+        return new_moves
+
+    def test_1(self):
```

Comment:
Please name these tests better & possibly add a one-line doc block for each explaining a bit more. Both the test name and first line of docblock show up in the tests, which is key to diagnosing issues later.